static/js/composebox_typeahead.js

```diff
@@ -47,15 +47,18 @@ export let emoji_collection = [];
 export function update_emoji_data() {
     emoji_collection = [];
     for (const emoji_dict of emoji.emojis_by_name.values()) {
+        const {reaction_type} = emoji.get_emoji_details_by_name(emoji_dict.name);
         if (emoji_dict.is_realm_emoji === true) {
             emoji_collection.push({
+                reaction_type,
                 emoji_name: emoji_dict.name,
                 emoji_url: emoji_dict.url,
                 is_realm_emoji: true,
             });
         } else {
             for (const alias of emoji_dict.aliases) {
                 emoji_collection.push({
+                    reaction_type,
```

Comment:
Specifically, I was thinking that the first commit would be just the changes in this file plus the test changes that doing so requires to it (which I think is like 75% of the PR). And then the second commit that actually does stuff would end up a lot more readable as a result.

Reply:
yes, I understand. I've made the revisions as you suggested. It has now been split into two commits